canonicals.js

```diff
@@ -0,0 +1,53 @@
+var APP = APP || {};
```

Comment:
If you're using modules already, why not just go with a solution like requirejs or commonjs and be done with it? (Not critical though)